lib/matplotlib/backend_bases.py

```diff
@@ -1629,6 +1632,35 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    @classmethod
+    @functools.lru_cache(1)
```

Comment:
Won't this cache be shared across all sub-classes?

Reply:
Sorry, indeed it should be `lru_cache()` (an unbounded cache).

The lru_cache(1) comes from mplcairo which did this just once with the list of hardcoded mappings just for mplcairo (as it predates the required_interactive_framework attribute).